Misc fixes on `cargo vendor-filterer`

cathay4t opened the pull request:
* Also fix SONAME on non-x86_64 linux build target.
* Store cargo vendor filterer into `Cargo.toml`.
* Guide user to run `cargo install cargo-vendor-filterer` when `cargo vendor-filterer` failed.

cgwalters commented on Makefile:
```diff
@@ -144,7 +144,9 @@ dist: manpage $(SPEC_FILE) $(CLIB_HEADER)
 		cd rust; \
 		cargo vendor-filterer \
 			--platform x86_64-unknown-linux-gnu \
```
I'd recommend the declarative spec in `Cargo.toml` BTW and also update to use `tier = "2"`, xref e.g. https://github.com/containers/bootc/blob/41ebc7f019e425b7bf43f2e6307788b34c4e2046/Cargo.toml#L12

cathay4t replied:
Thanks for the suggestion! Will try

cathay4t committed:
clib: Fix SONAME for all linux build target

Previously, we only fix SONAME for clib on `x86_64-unknown-linux-gnu`
target, we should do that on all linux build target. Hence changed to

    [target.'cfg(target_os="linux")']

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
Cargo.toml: Store cargo-vendor-filterer in config

Instruct cargo-vendor-filterer to only vendor crates for linux tier1+2
platform.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
Makefile: Provide better error message for cargo vendor-filterer

Guide user to run `cargo install cargo-vendor-filterer` when
`cargo vendor-filterer` failed.

Signed-off-by: Gris Ge <fge@redhat.com>
